Ienngrate Producer API

Mifaki opened the pull request:
### Changes Made

- Integrate all of prodcuer related usecase

### Bug (Server Related)
When updating episode, it will only update the status because right now it just use the mcr controller for updating it status

```typescript
    public function update(Episode $episode, Request $request): \Illuminate\Http\Response
    {
        $validated = $request->validate([
            'episode_id' => ['required'],
            'status' => ['required', Rule::enum(StatusEnum::class)],
        ]);
        $episode->status = $validated['status'];
        $episode->save();
        return response(status: 200);
    }
``` 

### UI
![image](https://github.com/user-attachments/assets/3903c2a1-3484-4878-82e7-0e8d6506e28b)
![image](https://github.com/user-attachments/assets/15e765e6-0fcd-4bd8-a111-3ebf66754a26)
![image](https://github.com/user-attachments/assets/ba3ed6b9-d744-45e7-bb8c-de276c02e838)
![image](https://github.com/user-attachments/assets/613ef119-075a-4a2e-b403-288a5acdb890)

Mifaki committed:
`feat(producer)`: Integrate Producer Page

Mifaki commented:
@mirzahilmi any update on this pr?

mirzahilmi commented:
> @mirzahilmi any update on this pr?

sorry, i was lately busy with my paperwork. I'll do the review tomorrow early or this late night if i have the time

mirzahilmi commented:
@Mifaki where the numbering of these episode from?

![image](https://github.com/user-attachments/assets/471cd932-05d6-4fde-b1f5-59368bdf9a9a)

its quite confusing when i deleted an episode, the ordering still the same, i.e i deleted episode 1, it realod and episode 1 is still there, but the total of the episodes does reduced

Mifaki commented:
> @Mifaki where the numbering of these episode from?
> 
> ![image](https://private-user-images.githubusercontent.com/118604764/402263066-471cd932-05d6-4fde-b1f5-59368bdf9a9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3MjQzMzAsIm5iZiI6MTczNjcyNDAzMCwicGF0aCI6Ii8xMTg2MDQ3NjQvNDAyMjYzMDY2LTQ3MWNkOTMyLTA1ZDYtNGZkZS1iMWY1LTU5MzY4YmRmOWE5YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExMlQyMzIwMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNjA2MjgxZWQ5MTI2Yzg2NzJjOWIxZDczYzM0MWYzMjk5N2FjZWYzOGJmYTVmYzE2YTA0NzI3MjU4ZGU1MjE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.tqAWsRJRyKDceMLDLuGOFzecJ6iLfDvhfsjxqPEEXm8)
> 
> its quite confusing when i deleted an episode, the ordering still the same, i.e i deleted episode 1, it realod and episode 1 is still there, but the total of the episodes does reduced

ah yeah mb, i think im just using index for the epsiode number, will fix it right away

Mifaki commented:
> @Mifaki where the numbering of these episode from?
> 
> ![image](https://private-user-images.githubusercontent.com/118604764/402263066-471cd932-05d6-4fde-b1f5-59368bdf9a9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3MjQzMzAsIm5iZiI6MTczNjcyNDAzMCwicGF0aCI6Ii8xMTg2MDQ3NjQvNDAyMjYzMDY2LTQ3MWNkOTMyLTA1ZDYtNGZkZS1iMWY1LTU5MzY4YmRmOWE5YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExMlQyMzIwMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNjA2MjgxZWQ5MTI2Yzg2NzJjOWIxZDczYzM0MWYzMjk5N2FjZWYzOGJmYTVmYzE2YTA0NzI3MjU4ZGU1MjE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.tqAWsRJRyKDceMLDLuGOFzecJ6iLfDvhfsjxqPEEXm8)
> 
> its quite confusing when i deleted an episode, the ordering still the same, i.e i deleted episode 1, it realod and episode 1 is still there, but the total of the episodes does reduced

sorry for the late response, but the episode numbers are designed to always maintain sequential order (1, 2, 3, etc.) regardless of deletions. This is intentional to keep the episode ordering - if episode 2 is deleted, the former episode 3 becomes the new episode 2, and so on. This way, there are never any gaps in the episode numbers, which could be confusing (like having episodes 1, 3, 4 because 2 was deleted).

mirzahilmi commented:
@Mifaki is there any way to change this input to be date only?

![image](https://github.com/user-attachments/assets/a6c59cff-87c8-4d80-9447-85571b4a160e)

mirzahilmi commented:
@Mifaki is there any way to change this input to take date value only?

![WhatsApp Image 2025-01-15 at 14 55 49](https://github.com/user-attachments/assets/e4a8b9b6-9bf6-460d-a7d1-6b107384763b)

Mifaki committed:
feat: add datepicker component

Mifaki commented:
@mirzahilmi i've created a ``datepicker`` component as your request to handle the date related field

<img width="1710" alt="image" src="https://github.com/user-attachments/assets/80fee5b5-a3bc-4a74-808b-c787c121f984" />

mirzahilmi committed:
Merge branch 'dev' into feat/producer-integrate

mirzahilmi commented:
perfect it works as expected. Could you pull the new changes from this branch? i added the api to update the whole episode resource on #36. Its merged onto the `dev` branch but i accidentally already merged it to this branch also because i was resolving the conflict not knowing it'll be commited to this PR, im sorry for that.

Mifaki committed:
feat(producer): integrate producer update api

Mifaki commented:
@mirzahilmi just update the pr with the new update episode

Mifaki committed:
feat(producer): change editForm tp using datepicker

mirzahilmi commented:
@Mifaki there's a bug for the create episode submission, i believe it doesnt actually send the request to the backend? could you please check it for me, thanks.

[Screencast From 2025-01-29 14-28-08.webm](https://github.com/user-attachments/assets/98f546bb-9dc9-4748-8e5c-e3eab166aa10)

Mifaki commented:
hi @mirzahilmi the form actually hit the api and create a program, your case happened because an error but the dialog is closing before displaying the message :), i've try to recreate your case and as you can see in the video below if it doesn't encounter any error it will work just fine

https://github.com/user-attachments/assets/06b76b5b-d484-4a8c-949d-5d1496b80221

ps: fixed with the latest commit

Mifaki committed:
fix(producer): prevent close dialog on error

mirzahilmi commented:
great

> hi @mirzahilmi the form actually hit the api and create a program, your case happened because an error but the dialog is closing before displaying the message :), i've try to recreate your case and as you can see in the video below if it doesn't encounter any error it will work just fine
> 
>  producer.mov 
> ps: fixed with the latest commit

great! now it works as intended. I guess we can move onto the next feature (if theres any) and Kak Aulia can do more thorough testing & quality check. Great job! @Mifaki

mirzahilmi reviewed (approved):
LGTM
